### PR TITLE
Validate prow labels, and don't hard-code prow.k8s.io

### PR DESCRIPTION
**Pull request by krzyzacy**
should fix https://github.com/kubernetes/test-infra/issues/8656

/area prow
/assign @stevekuznetsov @cjwagner

### CODE REVIEW

**Comment by krzyzacy** [comment]
/assign @fejta 
as you've reviewed https://github.com/kubernetes/test-infra/pull/8415

File: prow/pjutil/pjutil.go [hunk 1]
```diff
@@ -23,36 +23,52 @@ import (
...
 	"k8s.io/test-infra/prow/kube"
 )
 
 const (
-	jobNameLabel = "prow.k8s.io/job"
-	jobTypeLabel = "prow.k8s.io/type"
-	orgLabel     = "prow.k8s.io/refs.org"
-	repoLabel    = "prow.k8s.io/refs.repo"
-	pullLabel    = "prow.k8s.io/refs.pull"
+	defaultProw = "prow.k8s.io"
```
Comment (stevekuznetsov) [inline prow/pjutil/pjutil.go]: Uhhh this is the same as the API Group for a ProwJob -- not something to do with your website or deployment. This should be consistent and I would not consider this something that should be configured

Comment (stevekuznetsov) [reply prow/pjutil/pjutil.go]: The same way that people do not configure the API group for the ProwJob CRD

File: prow/pjutil/pjutil.go [hunk 2]
```diff
@@ -23,36 +23,52 @@ import (
...
 		}
 	}
 	for key, value := range labels {
 		allLabels[key] = value
 	}
+
+	// let's validate labels
+	for key, value := range allLabels {
+		if errs := validation.IsValidLabelValue(value); len(errs) > 0 {
+			delete(allLabels, key)
```
Comment (stevekuznetsov) [inline prow/pjutil/pjutil.go]: Is there not something to escape a value to make it a valid label? I don't want to lose UX here

File: prow/pjutil/pjutil.go (hunk 1 above)
Comment (BenTheElder) [reply prow/pjutil/pjutil.go]: Yeah, this is just a namespaced annotation. I don't think we need this change.

Comment (krzyzacy) [reply prow/pjutil/pjutil.go]: reverted

File: prow/pjutil/pjutil.go (hunk 2 above)
Comment (krzyzacy) [reply prow/pjutil/pjutil.go]: IMO we have/or should have presubmit to validate labels in some presubmit to ensure they are valid in the config :-)

Comment (krzyzacy) [reply prow/pjutil/pjutil.go]: I made it try to use basename first before delete the key

**Comment by stevekuznetsov** [comment]
Lacking any better escaping functionality, this seems fine for me

/lgtm

**Pull request merged**